runtime/interpreter/src/interpreter.cpp

```diff
@@ -425,7 +426,7 @@ int handleJumpEq(ShrimpVM *vm)
     auto rs_idx = instr.getRs();
     auto offset = instr.getJumpOffset();
 
-    auto eq = vm->acc().getValue() == frame.getReg(rs_idx).getValue();
+    auto eq = bit::getValue<int32_t>(vm->acc().getValue()) == bit::getValue<int32_t>(frame.getReg(rs_idx).getValue());
```

Comment:
Maybe compare operands as is?
Such approach could work if all integral operations results are sign-extended to register value

Reply:
Reverted